lib/presentation/widget/genres_row.dart

```diff
@@ -17,7 +17,7 @@ class Genres extends StatefulWidget {
 }
 
 class _GenresState extends State<Genres> {
-  final GenreRepository genreRepository = GenreRepository();
+  final GenreRepositoryImp genreRepository = GenreRepositoryImp();
```

Comment:
This should be `final IGenreRepository movieRepository = GenreRepositoryImp();`. 
The presentation layer relies on interfaces from the domain layer!

Reply:
done